sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2ScanExec.scala

```diff
@@ -95,21 +77,29 @@ case class DataSourceV2ScanExec(
           sparkContext.getLocalProperty(ContinuousExecution.EPOCH_COORDINATOR_ID_KEY),
           sparkContext.env)
         .askSync[Unit](SetReaderPartitions(readerFactories.size))
-      new ContinuousDataSourceRDD(sparkContext, sqlContext, readerFactories)
-        .asInstanceOf[RDD[InternalRow]]
-
-    case r: SupportsScanColumnarBatch if r.enableBatchRead() =>
-      new DataSourceRDD(sparkContext, batchReaderFactories).asInstanceOf[RDD[InternalRow]]
-
+      if (readerFactories.exists(_.dataFormat() == DataFormat.COLUMNAR_BATCH)) {
+        throw new IllegalArgumentException(
+          "continuous stream reader does not support columnar read yet.")
```

Comment:
cc @jose-torres do you know what's missing for this?

Reply:
I don't, because I'm not really sure how it works in the batch case. How does it work to do

new DataSourceRDD(sparkContext, batchReaderFactories).asInstanceOf[RDD[InternalRow]]

when the type parameter of batchReaderFactories doesn't match InternalRow?

Reply:
Then the missing piece is codegen. This is difficult because the continuous stream reader does a lot of auxiliary work, so I don't know if it will happen in the near future.

Reply:
I've thought about this further. Shouldn't it be trivial to write a wrapper that simply converts a DataReader[ColumnarBatch] to a DataReader[InternalRow]? If so then we can easily support it after the current PR.